### ISSUE TITLE
Unable to use plugins in 0.920

### ISSUE THREAD

**Issue by Apakottur** [python/mypy#11772]
**Bug Report**

I am using mypy plugins for libraries like [pydantic](https://github.com/samuelcolvin/pydantic) and [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

Attempting to upgrade mypy from 0.910 to 0.920 breaks when using any of those plugins.

**To Reproduce**

Create the two following files:

```python
# test.py
def func(var: int) -> int:
    return var
````

```ini
# mypy.ini
[mypy]
plugins=pydantic.mypy
```

Install `pydantic` and `mypy`, and run `mypy --config mypy.ini test.py`

**Expected Behavior**

No issues are found.

**Actual Behavior**

I am getting the following error:
```plain
mypy.ini:2: error: Error importing plugin "pydantic.mypy": cannot import name 'TypeVarDef' from 'mypy.types' (***/python3.10/site-packages/mypy/types.cpython-310-x86_64-linux-gnu.so)
Found 1 error in 1 file (errors prevented further checking)
```

**Your Environment**

- Mypy version used: `0.920`
- Python version used: `3.10.1`

**Comment by arnimarj** [comment]
An issue has been created for `pydantic`: https://github.com/samuelcolvin/pydantic/issues/3528
